diff --git a/train.py b/train.py
index a3838f6..f23f2ec 100644
--- a/train.py
+++ b/train.py
@@ -177,6 +177,8 @@ class CausalSelfAttention(nn.Module):
             ).transpose(1, 2)
 
         query, key, value = map(split_heads, (query, key, value))
+        query = F.rms_norm(query, (self.head_dim,))
+        key = F.rms_norm(key, (self.head_dim,))
         query, key = self.rope(query, key)
         attention = F.scaled_dot_product_attention(
             query,
