diff --git a/train.py b/train.py
index a3838f6..e17833c 100644
--- a/train.py
+++ b/train.py
@@ -211,7 +211,7 @@ class TransformerBlock(nn.Module):
         )
         self.mlp = nn.Sequential(
             nn.Linear(config.model_width, config.mlp_width, bias=False),
-            nn.GELU(),
+            nn.SiLU(),
             nn.Linear(config.mlp_width, config.model_width, bias=False),
         )
         self.dropout = nn.Dropout(config.dropout)
